skimage/util/compare.py

```diff
@@ -1,27 +1,100 @@
+import functools
+import warnings
+from itertools import product
+
 import numpy as np
 
 from .dtype import img_as_float
-from itertools import product
 
 
-def compare_images(image1, image2, method='diff', *, n_tiles=(8, 8)):
+def _rename_image_params(func):
+    wm_images = (
+        "Since version 0.23, the two input images are named `image0` and "
+        "`image1` (instead of `image1` and `image2`, respectively). Please use "
+        "`image0, image1` to avoid this warning for now, and avoid an error "
+        "from version 0.25 onwards."
+    )
+
+    wm_method = (
+        "Starting in version 0.25, all arguments following `image0, image1` "
+        "(including `method`) will be keyword-only. Please pass `method=` "
+        "in the function call to avoid this warning for now, and avoid an error "
+        "from version 0.25 onwards."
+    )
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        # Turn all args into kwargs
+        for i, (value, param) in enumerate(
+            zip(args, ["image0", "image1", "method", "n_tiles"])
+        ):
+            if i >= 2:
+                warnings.warn(wm_method, category=FutureWarning)
+            if param in kwargs:
+                raise ValueError(
+                    f"{param} passed both as positional and keyword argument."
+                )
+            else:
+                kwargs[param] = value
+        args = tuple()
+
+        # Account for `image2` if given
+        if "image2" in kwargs.keys():
+            warnings.warn(wm_images, category=FutureWarning)
+
+            # Safely move `image2` to `image1` if that's empty
+            if "image1" in kwargs.keys():
+                # Safely move `image1` to `image0`
+                if "image0" in kwargs.keys():
+                    raise ValueError(
+                        "Three input images given; please use only `image0` "
+                        "and `image1`."
+                    )
+                kwargs["image0"] = kwargs.pop("image1")
+            kwargs["image1"] = kwargs.pop("image2")
+
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+@_rename_image_params
+def compare_images(image0, image1, method='diff', *, n_tiles=(8, 8)):
     """
     Return an image showing the differences between two images.
 
     .. versionadded:: 0.16
 
     Parameters
     ----------
-    image1, image2 : ndarray, shape (M, N)
-        Images to process, must be of the same shape.
+    image0 : ndarray, shape (M, N)
+        First input image.
+
+        .. versionadded:: 0.23
+    image1 : ndarray, shape (M, N)
+        Second input image. Must be of the same shape as `image0`.
+
+        .. versionchanged:: 0.23
+            `image1` changed from being the name of the first image to that of
+            the second image.
```

Comment:
Thanks for documenting the change in such detail! Though, what do you think about simplifying it somewhat so it's easier to grok?

```suggestion
    image0, image1 : ndarray, shape (M, N)
        Images to process, must be of the same shape.

        .. versionchanged:: 0.23
            `image1` and `image2` where renamed to `image0` and `image1`
            respecitively.
```